src/controllers/assistant.js

```diff
@@ -45,91 +46,169 @@ function AssistantController(context) {
+   * @param requestContext.pathInfo Object containing the request path and headers
+   * @param requestContext.pathInfo.headers Object containing the request headers
+   * @param requestContext.pathInfo.headers.x-api-key String api-key
+   * @param requestContext.pathInfo.headers.x-gw-ims-org-id String ims-org-id
+   * @param requestContext.env Object containing the environment variables (secrets, etc.)
+   * @param requestContext.env.ASSISTANT_CONFIGURATION Object containing the assistant configuration
+   * @param requestContext.env.ASSISTANT_PROMPTS Object containing the assistant prompts
+   * @returns the validated and parsed request context data
+   */
+  function parseRequestContext(requestContext) {
+    // Validate input data.
+    if (!requestContext) {
+      throw new ErrorWithStatusCode(
+        'Invalid request: missing request context.',
+        STATUS.BAD_REQUEST,
+      );
     }
+    if (!isObject(requestContext)
+      || !requestContext.data
+      || !requestContext.attributes
+      || !requestContext.pathInfo
+    ) {
+      throw new ErrorWithStatusCode(
+        'Invalid request: invalid request context format.',
+        STATUS.BAD_REQUEST,
+      );
+    }
+
+    const { data, pathInfo: { headers }, attributes } = requestContext;
+
+    validateRequestData(data);
+    validateRequestAttributes(requestContext.attributes, headers['x-gw-ims-org-id']);
+    const contextEnv = processAndValidateContextEnv(context.env);
 
-    const { data: { command, prompt, options }, attributes } = requestContext;
-    const { firefallArgs = {} } = commandConfig[command] || {};
-    const { imageUrl, ...otherOptions } = options || {};
+    // Validation complete. Return the parsed request data.
+    const { command, prompt, options } = data;
+    const mergedPrompt = mergePrompt(
+      contextEnv.ASSISTANT_PROMPTS,
+      command,
+      {
+        content: options.htmlContent,
+        selector: options.selector,
+        pattern: prompt,
+      },
+    );
+
+    const { firefallArgs = {} } = commandConfig[command];
     const { authInfo: { profile: apikeyProfile } } = attributes;
-    const callerImsOrgId = apikeyProfile?.getImsOrgId() ?? requestContext.pathInfo.headers['x-gw-ims-org-id'];
-    const requestData = {
+    // TODO: remove header check when on Prod, and/or when api-key's have that association.
+    const callerImsOrgId = apikeyProfile?.getImsOrgId()
+      ?? requestContext.pathInfo.headers['x-gw-ims-org-id'];
+
+    return {
       env: {
-        ...context.env,
-        ...assistantConfiguration,
-        FIREFALL_API_KEY: assistantConfiguration.IMS_CLIENT_ID,
+        ...contextEnv,
         FIREFALL_IMS_ORG_ID: callerImsOrgId,
       },
       ...firefallArgs,
       command,
-      prompt,
-      imageUrl,
+      prompt: mergedPrompt,
+      ...options,
       importApiKey: requestContext.pathInfo.headers['x-api-key'],
       apiKeyName: apikeyProfile?.getName(),
-      otherOptions,
```

Comment:
We could be more selective with what is in the final requestData (for instance, don't need all the ...options), but what isn't required is ignored, so thought I'd leave it in for now for simplicity.